paimon-flink/paimon-flink-cdc/src/main/java/org/apache/paimon/flink/action/cdc/postgres/PostgresActionUtils.java

```diff
@@ -0,0 +1,199 @@
+                .hostname(postgresConfig.get(PostgresSourceOptions.HOSTNAME))
+                .port(postgresConfig.get(PostgresSourceOptions.PG_PORT))
+                .database(postgresConfig.get(PostgresSourceOptions.DATABASE_NAME))
+                .schemaList(schemaList)
+                .tableList(tableList)
+                .slotName(postgresConfig.get(PostgresSourceOptions.SLOT_NAME))
+                .username(postgresConfig.get(PostgresSourceOptions.USERNAME))
+                .password(postgresConfig.get(PostgresSourceOptions.PASSWORD));
+
+        // use pgoutput for PostgreSQL 10+
+        postgresConfig
+                .getOptional(PostgresSourceOptions.DECODING_PLUGIN_NAME)
+                .ifPresent(sourceBuilder::decodingPluginName);
+
+        // PostgreSQL CDC using increment snapshot, splitSize is used instead of fetchSize (as in
+        // JDBC
+        // connector). splitSize is the number of records in each snapshot split.
+        postgresConfig
+                .getOptional(PostgresSourceOptions.SCAN_INCREMENTAL_SNAPSHOT_CHUNK_SIZE)
+                .ifPresent(sourceBuilder::splitSize);
+        postgresConfig
+                .getOptional(PostgresSourceOptions.CONNECT_TIMEOUT)
+                .ifPresent(sourceBuilder::connectTimeout);
+        postgresConfig
+                .getOptional(PostgresSourceOptions.CONNECT_MAX_RETRIES)
+                .ifPresent(sourceBuilder::connectMaxRetries);
+        postgresConfig
+                .getOptional(PostgresSourceOptions.CONNECTION_POOL_SIZE)
+                .ifPresent(sourceBuilder::connectionPoolSize);
+        postgresConfig
+                .getOptional(PostgresSourceOptions.HEARTBEAT_INTERVAL)
+                .ifPresent(sourceBuilder::heartbeatInterval);
+
+        String startupMode = postgresConfig.get(PostgresSourceOptions.SCAN_STARTUP_MODE);
+
+        if ("initial".equalsIgnoreCase(startupMode)) {
+            sourceBuilder.startupOptions(StartupOptions.initial());
+        } else if ("earliest-offset".equalsIgnoreCase(startupMode)) {
+            sourceBuilder.startupOptions(StartupOptions.earliest());
+        } else if ("latest-offset".equalsIgnoreCase(startupMode)) {
+            sourceBuilder.startupOptions(StartupOptions.latest());
+        } else if ("specific-offset".equalsIgnoreCase(startupMode)) {
+            String file =
+                    postgresConfig.get(PostgresSourceOptions.SCAN_STARTUP_SPECIFIC_OFFSET_FILE);
+            Integer pos =
+                    postgresConfig.get(PostgresSourceOptions.SCAN_STARTUP_SPECIFIC_OFFSET_POS);
+            sourceBuilder.startupOptions(StartupOptions.specificOffset(file, pos));
+        } else if ("timestamp".equalsIgnoreCase(startupMode)) {
+            sourceBuilder.startupOptions(
+                    StartupOptions.timestamp(
+                            postgresConfig.get(
+                                    PostgresSourceOptions.SCAN_STARTUP_TIMESTAMP_MILLIS)));
+        }
```

Comment:
See ：https://github.com/ververica/flink-cdc-connectors/blob/e3d6c7e0aa9b9b8e573311b879f8854d7c88f88f/flink-cdc-connect/flink-cdc-source-connectors/flink-connector-postgres-cdc/src/main/java/com/ververica/cdc/connectors/postgres/table/PostgreSQLTableFactory.java#L207

![1705591868234_3D7D86A9-CCCA-435c-BE61-2B716751BA11](https://github.com/apache/incubator-paimon/assets/45089228/29e4e162-9281-4ee6-84a9-56a5e4688c52)

Some start modes are not implemented in flink cdc, please fix it.

Reply:
Thanks @TaoZex, done.